### ISSUE TITLE
How to change LayoutLMv3 to a LayoutLMv3 XLM (i.e. LayoutXLM-like model)

### ISSUE THREAD

**Issue by piegu** [NielsRogge/Transformers-Tutorials#253]
Hi @NielsRogge,

I plan to finetune a LayoutXLM large like model. Why "like model"? Because until now, Microsoft did not release LayoutXLM large mas only a version base.

As I want to train a version large on documents of another language as English (I have a huge labeled documents dataset), I need to change the LayoutLMv3 large tokenizer from RoBERTa to XLM RoBERTa.

I understand from the page 9 of the paper [LayoutLMv3: Pre-training for Document AI with Unified Text and Image Masking](https://arxiv.org/abs/2204.08387), that this change was done for the LayoutLMv3 in Chinese:

`For the multimodal Transformer encoder along with the text embedding layer, LayoutLMv3-Chinese is initialized from the pre-trained weights of XLM-R [7]. We randomly initialized the rest model parameters.`

From this citation, I understand that Microsoft copied/pasted the embeddings weigths of XLM-RoBERTa to the ones of LayoutLMv3. 

1. How to do that?
2. and how to change also the LayoutLMv3Tokenizer to the XLM-RoBERTa one?

About the question 2, it would be as simple as what you did in your notebook "[Create_LiLT_+_XLM_RoBERTa_base.ipynb](https://github.com/NielsRogge/Transformers-Tutorials/blob/master/LiLT/Create_LiLT_%2B_XLM_RoBERTa_base.ipynb)" for a LiLT model with XLM-RoBERTa?

```
from transformers import AutoTokenizer
tokenizer = AutoTokenizer.from_pretrained("xlm-roberta-base")
tokenizer.push_to_hub("nielsr/lilt-xlm-roberta-base")
```

Thank you.

**Comment by NielsRogge** [comment]
To create a LayoutXLM-large, you can copy paste the weights from [xlm-roberta-large](https://huggingface.co/xlm-roberta-large) into the model. You can do that by converting the weights as done in any conversion script, like [this one](https://github.com/huggingface/transformers/blob/main/src/transformers/models/beit/convert_beit_unilm_to_pytorch.py). You basically need to load the original state_dict and then manipulate the dictionary to make it have the appropriate names for the parameters names in the model that you'd like to equip the weights with.

To use a tokenizer, you can use the LayoutXLMTokenizer (and LayoutXLMProcessor) which is already available in 🤗 Transformers.

I don't think it's possible to leverage LiLT for large models, as LiLT only comes in a base-sized version.

**Comment by NielsRogge** [comment]
@philmas the best way is to combine Lilt with a Dutch roberta-base model. 

For Dutch, the best roberta-base model is currently this one: https://huggingface.co/DTAI-KULeuven/robbert-2023-dutch-base.

Hence please follow this guide to combine it with LiLT: https://github.com/jpWang/LiLT?tab=readme-ov-file#or-generate-your-own-checkpoint-optional.

**Comment by NielsRogge** [comment]
Usuallly I recommend starting with 100, but as always with deep learning, the more the better

**Comment by LuckaGianvechio** [comment]
Hello @NielsRogge, I am currently trying reproduce pretraining of a LayoutLMv3 model in a Brazilian Portuguese dataset. 

Is there a way to initialize the model's weights with another BERT based model called Bertimbau (since it isn't a roberta-base model) ? I have currently initialized only the word embeddings layer with Bertimbau's, and reimplemented the tokenization to work with Bertimbau's token ids, but I think I could achieve better results by initializing more parameters with it. 

Also, I haven't found where to provide the model information about witch image patches to mask (for the MIM and WPA tasks) in the hugging face transformers modeling_layoutlmv3.py code. Do I need to implement this by hand in LayoutLMv3Model class or is there something already implemented?

**Comment by alan-insam** [comment]
> Hello [@NielsRogge](https://github.com/NielsRogge), I am currently trying reproduce pretraining of a LayoutLMv3 model in a Brazilian Portuguese dataset.
> 
> Is there a way to initialize the model's weights with another BERT based model called Bertimbau (since it isn't a roberta-base model) ? I have currently initialized only the word embeddings layer with Bertimbau's, and reimplemented the tokenization to work with Bertimbau's token ids, but I think I could achieve better results by initializing more parameters with it.
> 
> Also, I haven't found where to provide the model information about witch image patches to mask (for the MIM task) in the hugging face transformers modeling_layoutlmv3.py code. Do I need to implement this by hand in LayoutLMv3Model class or is there something already implemented?
> 
> Thanks in advance.

Hey, have you been able to finetune LayoutLMv3 on your Portugese dataset?
